sql/core/src/main/scala/org/apache/spark/sql/SparkSessionExtensions.scala

```diff
@@ -113,6 +113,24 @@ class SparkSessionExtensions {
     columnarRuleBuilders += builder
   }
 
+  private[this] val resolutionHintBuilders = mutable.Buffer.empty[RuleBuilder]
+
+  /**
+    * Build the analyzer resolution `Hint`s using the given [[SparkSession]].
+    */
+  private[sql] def buildResolutionHint(session: SparkSession): Seq[Rule[LogicalPlan]] = {
+    resolutionHintBuilders.map(_.apply(session))
+  }
+
+  /**
+    * Inject an analyzer resolution `Hint` builder into the [[SparkSession]]. These analyzer
+    * rules will be executed as part of the resolution phase of analysis.
+    */
+  def injectResolutionHint(builder: RuleBuilder): Unit = {
```

Comment:
Did you try to use `def injectResolutionRule(builder: RuleBuilder)`? Is it insufficient for your use-case?

Reply:
@dongjoon-hyun  
Since for Hint , it will call **ResolveHints.RemoveAllHintsafter** it handle all Spark's hint ,.
So add Hint rule in
` def injectResolutionRule(builder: RuleBuilder)`
won't work since all hint has been cleared, That's why I add a new method for hint.